index.js

```diff
@@ -0,0 +1,229 @@
+// Resume Builder
+const fs = require('fs');
+const readline = require('readline');
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+
+let resume = {
+    "basics": {
+        "name": "",
+        "label": "",
+        "picture": "",
+        "email": "",
+        "phone": "",
+        "website": "",
+        "summary": ""
+      },
+      "work": [],
+      "education": [],
+      "skills": [],
+};
+let fileName = 'resume.json';
+
+console.log('==============================\nWelcome to Resume Builder!\n==============================');
+
+// Check if fileName is given (if not, use default)
+if (process.argv[2]) {
+    fileName = process.argv[2];
+    showCommands();
+} else {
+    showCommands();
+}
+
+// Read .json file and add to resume
+fs.readFile(fileName, (err, data) => {
+    if (typeof data !== 'undefined') {
+        resume = JSON.parse(data);
+    }
+});
+
+function showCommands() {
+    // Show command options after every request is completed
+    rl.question(`(v) View • (a) Add • (d) Delete • (s) Save • (q) Quit\n`, answer => {
+        // View resume
+        if (answer === 'v') {
+            const categories = Object.entries(resume);
+            categories.forEach(category => console.log(category));
+            showCommands();
+        // Add resume item
+        } else if (answer === 'a') {
+            const getCategory = () => {
+                rl.question('(b) Basics • (w) Work • (e) Education • (s) Skills\n', category => {
+                    switch(category) {
+                        case 'b':
+                            const getBasic = () => {
+                                rl.question('Please select a field: name • label • picture • email • phone • summary\n', field => {
+                                    field === 'name' || field === 'label' || field === 'picture' || field === 'email' || field === 'phone' || field === 'summary' ? 
+                                    rl.question(`Enter ${field}:\n`, answer => {
+                                        resume['basics'][field] = answer;
+                                        showCommands();
+                                    })
+                                    : 
+                                    console.log("Please enter a valid field.");
+                                    getBasic();
+                                });
+                            };
+                            
+                            getBasic();
+                            break;
+                        case 'w':
+                            let job = {};
+
+                            const getCompany = answer => {
```

Comment:
I split this code into a separate function for this case along with your other cases. This will make the code base easier to maintain and debug. Also try to leverage the structure that you've already defined above in the resume variable by looping though it to determine which questions and functions to call.

Reply:
Thanks for the tip!!! :)